data-prepper-plugins/geoip-processor/src/main/java/org/opensearch/dataprepper/plugins/processor/GeoIPProcessor.java

```diff
@@ -52,14 +55,17 @@ public class GeoIPProcessor extends AbstractProcessor<Record<Event>, Record<Even
    */
   @DataPrepperPluginConstructor
   public GeoIPProcessor(PluginSetting pluginSetting,
-                        final GeoIPProcessorConfig geoCodingProcessorConfig) {
+                        final GeoIPProcessorConfig geoCodingProcessorConfig,
+                        final GeoIpConfigSupplier geoIpConfigSupplier) {
     super(pluginSetting);
     this.geoIPProcessorConfig = geoCodingProcessorConfig;
     this.tempPath = System.getProperty("java.io.tmpdir")+ File.separator + TEMP_PATH_FOLDER;
     geoIPProcessorService = new GeoIPProcessorService(geoCodingProcessorConfig,tempPath);
     tagsOnSourceNotFoundFailure = geoCodingProcessorConfig.getTagsOnSourceNotFoundFailure();
     this.geoIpProcessingMatchCounter = pluginMetrics.counter(GEO_IP_PROCESSING_MATCH);
     this.geoIpProcessingMismatchCounter = pluginMetrics.counter(GEO_IP_PROCESSING_MISMATCH);
+    // TODO: use this config and clean up MaxMind service config from pipeline.yaml
+    geoIpServiceConfig = geoIpConfigSupplier.getGeoIpServiceConfig();
```

Comment:
I think that this code should want to get a `GeoIpService`. That is, we can have one underlying object for handling GeoIP requests.

By doing that, we'd be able to share the underlying database and structures across multiple instances of `GeoIPProcessor`. This can reduce the memory load in Data Prepper.

So, I think we'd want to pass into the constructor:

```
final GeoIpServiceProvider geoIpServiceProvider
```

And then call:

```
geoIpService = geoIpServiceProvider.getService();
```

Reply:
Yeah, I like the suggestion. This way only one GeoIP service is created and updated once per node. I will implement in a different PR, I will expose a `getGeoIpService` in the `GeoIpConfigSupplier` interface and it will create the service using config.

Reply:
I'm ok with creating in a new PR. But, if you go that route, I think we will want to remove the code that exposes the config outside of the extension. I think it will be unnecessary.